test/e2e/llmisvc/test_llm_inference_service.py

```diff
@@ -82,35 +89,121 @@ class TestCase:
+                "where the benefits of P/D disaggregation would strongly outweigh its complexities. "
+                "Conclude by speculating on emerging technologies or paradigms (e.g., serverless compute functions "
+                "directly interacting with object storage, in-memory disaggregation) that could further drive or "
+                "transform P/D disaggregation in cloud computing.",
+                max_tokens=2000,
+            ),
+            marks=[
+                pytest.mark.cluster_gpu,
+                pytest.mark.cluster_nvidia,
+                pytest.mark.cluster_nvidia_roce,
+            ],
+        ),
+        pytest.param(
+            TestCase(
+                base_refs=[
+                    "router-no-scheduler",
+                    "workload-single-cpu",
+                    "model-fb-opt-125m",
+                ],
+                prompt="What is KServe?",
+            ),
+            marks=[
+                pytest.mark.cluster_cpu,
+                pytest.mark.cluster_single_node,
+                pytest.mark.no_scheduler,
+            ],
+        ),
+        pytest.param(
+            TestCase(
+                base_refs=[
+                    "router-managed",
+                    "workload-simulated-dp-ep-cpu",
+                    "model-fb-opt-125m",
+                ],
+                prompt="This test simulates DP+EP that can run on CPU, the idea is to test the LWS-based deployment, "
+                "but without the resources requirements for DP+EP (GPUs and ROCe/IB).",
+            ),
+            marks=[pytest.mark.cluster_cpu, pytest.mark.cluster_multi_node],
+        ),
     ],
     indirect=["test_case"],
     ids=generate_test_id,
 )
 @log_execution
 def test_llm_inference_service(test_case: TestCase):
+    inject_k8s_proxy()
 
     kserve_client = KServeClient(
-        config_file=os.environ.get("KUBECONFIG", "~/.kube/config")
+        config_file=os.environ.get("KUBECONFIG", "~/.kube/config"),
+        client_configuration=client.Configuration(),
     )
 
     service_name = test_case.llm_service.metadata.name
 
     try:
         create_llmisvc(kserve_client, test_case.llm_service)
+        wait_for_llm_isvc_ready(
+            kserve_client, test_case.llm_service, test_case.wait_timeout
+        )
         wait_for_model_response(kserve_client, test_case, test_case.wait_timeout)
     except Exception as e:
         print(f"❌ ERROR: Failed to call llm inference service {service_name}: {e}")
         _collect_diagnostics(kserve_client, test_case.llm_service)
         raise
     finally:
         try:
-            delete_llmisvc(kserve_client, test_case.llm_service)
+            if os.getenv("SKIP_RESOURCE_DELETION", "False").lower() in (
+                "false",
+                "0",
+                "f",
+            ):
+                delete_llmisvc(kserve_client, test_case.llm_service)
```

Comment:
```suggestion
            skip = os.getenv("SKIP_RESOURCE_DELETION", "").lower() in {"true", "1", "t"}
            if skip:
                pytest.skip(f"Skipping resource deletion for {service_name}")
            delete_llmisvc(kserve_client, test_case.llm_service)
```

Or wrap in `maybe_delete_...`

Reply:
This seems not ideal as skips the test https://docs.pytest.org/en/stable/how-to/skipping.html#skipping-test-functions ?